Cannot build C++ example

I'm on macOS. After building and installing torchvision, I tried building the C++ example and failed.

`
/Users/martin/devel/vision/examples/cpp/hello_world/main.cpp:3:10: fatal error: 'torchvision/models/resnet.h' file not found
#include <torchvision/models/resnet.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/hello-world.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/hello-world.dir/all] Error 2
make: *** [all] Error 2
`

Adding

`target_include_directories(TorchVision::TorchVision INTERFACE "${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_INCLUDEDIR@")`

after

https://github.com/pytorch/vision/blob/190a5f8a32f9ae775d4379c55db7e2deb6eada6b/cmake/TorchVisionConfig.cmake.in#L34

or replacing the `target_link_libraries` command with

`
set_target_properties(TorchVision::TorchVision PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_INCLUDEDIR@"
  INTERFACE_LINK_LIBRARIES "torch;Python3::Python"
)
`

and explicitly setting the `INTERFACE_INCLUDE_DIRECTORIES` property seems to fix the problem.

--- a/examples/cpp/hello_world/CMakeLists.txt
+++ b/examples/cpp/hello_world/CMakeLists.txt
@@ -11,6 +11,6 @@ add_executable(hello-world main.cpp)
 # We now need to link the TorchVision library to our executable.
 # We can do that by using the TorchVision::TorchVision target,
 # which also adds all the necessary torch dependencies.
-target_compile_features(test_frcnn_tracing PUBLIC cxx_range_for)
+target_compile_features(hello-world PUBLIC cxx_range_for)
 target_link_libraries(hello-world TorchVision::TorchVision)
-set_property(TARGET test_frcnn_tracing PROPERTY CXX_STANDARD 14)
+set_property(TARGET hello-world PROPERTY CXX_STANDARD 14)

--- a/examples/cpp/hello_world/CMakeLists.txt
+++ b/examples/cpp/hello_world/CMakeLists.txt
@@ -11,4 +11,6 @@ add_executable(hello-world main.cpp)
 # We now need to link the TorchVision library to our executable.
 # We can do that by using the TorchVision::TorchVision target,
 # which also adds all the necessary torch dependencies.
+target_compile_features(test_frcnn_tracing PUBLIC cxx_range_for)
 target_link_libraries(hello-world TorchVision::TorchVision)
+set_property(TARGET test_frcnn_tracing PROPERTY CXX_STANDARD 14)

--- a/examples/cpp/hello_world/main.cpp
+++ b/examples/cpp/hello_world/main.cpp
@@ -1,5 +1,4 @@
 #include <iostream>
-
 #include <torchvision/models/resnet.h>
 
 int main()
@@ -15,7 +14,7 @@ int main()
 
   if (torch::cuda::is_available()) {
     // Move model and inputs to GPU
-    model.to(torch::kCUDA);
+    model->to(torch::kCUDA);
     auto gpu_in = in.to(torch::kCUDA);
     auto gpu_out = model->forward(gpu_in);
 

--- a/examples/cpp/hello_world/main.cpp
+++ b/examples/cpp/hello_world/main.cpp
@@ -12,4 +12,13 @@ int main()
   auto out = model->forward(in);
 
   std::cout << out.sizes();
+
+  if (torch::cuda::is_available()) {
+    // Move model and inputs to GPU
+    model.to(torch::kCUDA);
+    auto gpu_in = in.to(torch::kCUDA);
+    auto gpu_out = model->forward(gpu_in);
+
+    std::cout << gpu_out.sizes();
+  }
 }

--- a/examples/cpp/hello_world/main.cpp
+++ b/examples/cpp/hello_world/main.cpp
@@ -11,5 +11,5 @@ int main()
   auto in = torch::rand({1, 3, 10, 10});
   auto out = model->forward(in);
 
-  std::cout << out;
+  std::cout << out.sizes();
 }

--- a/packaging/build_cmake.sh
+++ b/packaging/build_cmake.sh
@@ -61,7 +61,7 @@ popd
 python setup.py develop
 
 # Trace, compile and run project that uses Faster-RCNN
-cd test/tracing/frcnn
+pushd test/tracing/frcnn
 mkdir build
 
 # Trace model
@@ -81,3 +81,21 @@ fi
 
 # Run traced program
 ./test_frcnn_tracing
+
+# Compile and run the CPP example
+popd
+cd examples/cpp/hello_world
+
+mkdir build
+cd build
+cmake .. -DTorch_DIR=$TORCH_PATH/share/cmake/Torch
+
+if [[ "$OSTYPE" == "msys" ]]; then
+    "$script_dir/windows/internal/vc_env_helper.bat" "$script_dir/windows/internal/build_cpp_example.bat"
+    cd Release
+else
+    make
+fi
+
+# Run CPP example
+./hello-world

--- a/packaging/windows/internal/build_cpp_example.bat
+++ b/packaging/windows/internal/build_cpp_example.bat
@@ -1,0 +1,3 @@
+@echo on
+set CL=/I"C:\Program Files (x86)\torchvision\include"
+msbuild "-p:Configuration=Release" hello-world.vcxproj

